core-primitives/types/src/lib.rs

```diff
@@ -42,16 +41,14 @@ pub type PalletString = String;
 
 pub use sp_core::{crypto::AccountId32 as AccountId, H256};
 
+pub use itp_sgx_runtime_primitives::types::*;
+
 // FIXME: Consolidate type definiton with #905.
-pub type Balance = u128;
 pub type AccountData = pallet_balances::AccountData<Balance>;
```

Comment:
I think the following types should also be moved to the runtime primitives:
- `AccountData`
- `AccountInfo`
- `SidechainBlockNumber`
- `SidechainTimestamp`

Reasoning: They belong to a (future) pallet, and heavily affect the runtime state.

And because you also moved the header, I guess moving the block type should make sense as well:
- `Block`
- `SignedBlock`
- `BlockHash`

I'm unsure about:
- `ShardIdentifier `

@clangenb What would be your idea? Everything used in the runtime, moved to these primitives? If so, then all the above listed types should be moved.

Reply:
I agree, all of them should be moved.

Sidechain stuff should ideally be in a sidechain module. I consider the following types in the sidechain domain:

```
SidechainBlockNumber
SidehainTimestamp
Block
SignedBlock
BlockHash
```

Reply:
Thank you both for the detailed notes, I move all of them.